presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2031,55 +2051,100 @@ public void run()
 
     private static class CreateTableOperation
     {
-        private final Table table;
+        private final Table newTable;
         private final PrincipalPrivileges privileges;
         private boolean done;
+        private final boolean ignoreExisting;
 
-        public CreateTableOperation(Table table, PrincipalPrivileges privileges)
+        public CreateTableOperation(Table newTable, PrincipalPrivileges privileges, boolean ignoreExisting)
         {
-            requireNonNull(table, "table is null");
-            checkArgument(getPrestoQueryId(table).isPresent());
-            this.table = table;
+            requireNonNull(newTable, "newTable is null");
+            checkArgument(getPrestoQueryId(newTable).isPresent());
+            this.newTable = newTable;
             this.privileges = requireNonNull(privileges, "privileges is null");
+            this.ignoreExisting = ignoreExisting;
         }
 
         public String getDescription()
         {
-            return format("add table %s.%s", table.getDatabaseName(), table.getTableName());
+            return format("add table %s.%s", newTable.getDatabaseName(), newTable.getTableName());
         }
 
         public void run(ExtendedHiveMetastore metastore)
         {
             try {
-                metastore.createTable(table, privileges);
+                metastore.createTable(newTable, privileges);
                 done = true;
             }
             catch (RuntimeException e) {
-                try {
-                    Optional<Table> remoteTable = metastore.getTable(table.getDatabaseName(), table.getTableName());
-                    // getPrestoQueryId(partition) is guaranteed to be non-empty. It is asserted in the constructor.
-                    if (remoteTable.isPresent() && getPrestoQueryId(remoteTable.get()).equals(getPrestoQueryId(table))) {
+                handleException(metastore, e);
+            }
+        }
+
+        private void handleException(ExtendedHiveMetastore metastore, RuntimeException exception)
+        {
+            try {
+                Optional<Table> existingTable = metastore.getTable(newTable.getDatabaseName(), newTable.getTableName());
+                existingTable.ifPresent(table -> {
```

Comment:
Can you change this back to `if (remoteTable.isPresent() {` please? When you have a multiple-line lambda like this, using `ifPresent` makes code harder to read.